server/api.js

```diff
@@ -12,14 +14,61 @@ router.get("/events", (req,res)=> {
 
 	// const eventQuery="SELECT meeting_location, meeting_start, meeting_end, meeting_address_1, meeting_city, meeting_postcode FROM events WHERE meeting_end >=current_date AND meeting_end <= current_date + INTERVAL '7 day'";
 	const evq = `SELECT * FROM events`;
-	
+
 	pool.query(evq)
 	.then((result)=>res.json(result))
 	.catch((error)=>{
 		console.error(error);
 		res.status(500).json(error);
 	});
-	
+});
+
+// endpoint for register form
+router.post("/register", async (req, res) => {
+	const { name, email, region, classNr, password } = req.body;
+	const role = req.body.role == "Admin" ? true : false;
+	const isLunchMaker = false;
+	const isLunchShopper = false;
+
+	//hashing the password
+	const saltRounds = 10;
+	const salt = await bcrypt.genSalt(saltRounds);
+	const encryptedPassword = await bcrypt.hash(password, salt);
+
+	//get the cohortId from cohort table
+	const getCohortId =
+		"Select id FROM cohort WHERE class_number = $1 AND region = $2";
+
+	const cohortId = await pool
+		.query(getCohortId, [classNr, region])
+		.then((result) => result.rows[0].id);
+
+	//check if email is already used
+	pool
+		.query("SELECT * FROM users WHERE user_email = $1", [email])
+		.then((result) => {
+			if (result.rows.length > 0) {
+				return res
+					.status(400)
```

Comment:
In this case status code must be 409, there is a conflict coz there is already existed email but 400 is bad request (example due to a bad syntax...etc)

Reply:
Same here